### PR TITLE
escape translations in pegasus

### PR DESCRIPTION
**Pull request by Hamms**
# Description

So, it turns out that sinatra does not escape HTML in template strings by default like Rails does, so all our pegasus i18n strings are vulnerable to HTML injection.

My proposed solution is to update both of the i18n methods we use in pegasus (`I18n.t` and `hoc_s`) to return escaped HTML, and then require template editors to explicitly unescape HTML in strings if they need HTML.

We should then in a future pass go through all strings that are being unescaped and update them to not require HTML (probably by updating them to use sanitized markdown).

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by Hamms** [comment]
Note that any calls to `i18n.t` or `hoc_s` that are:

1. using HTML in the string, and
2. programmatically assembling the key rather than using it directly

Will not be caught by this change, and will start rendering the escaped HTML. I'm open to suggestions as to whether we want to try and track down any of those calls in advance, or simply respond to them in they crop up.

File: pegasus/sites.v3/code.org/public/congrats.haml [hunk 1]
```diff
@@ -54,7 +54,7 @@ max_age: 60
         =TextRender.safe_markdown company['additional_actions_copy_t']
       -else
         =view :company_profile_additional_actions
-      = I18n.t :congratulations_volunteer
+      = CGI.escapeHTML(I18n.t(:congratulations_volunteer))
```
Comment (bethanyaconnor) [inline pegasus/sites.v3/code.org/public/congrats.haml]: Shouldn't this (and all the others) be `CGI.unescapeHTML`?

Comment (Hamms) [reply pegasus/sites.v3/code.org/public/congrats.haml]: d'oh

**Review by bethanyaconnor — APPROVED** [review]
Looks good! It might be worth giving other devs a heads up before this ships that if they see any escaped HTML it might be due to this change.

**Comment by Hamms** [comment]
Unit test failure appears to be related: 
image

**Comment by Hamms** [comment]
Dangit; looks like https://github.com/code-dot-org/code-dot-org/pull/32496/files#diff-015ffd15b25d6d102b984dc224e0ec54R69 actually affects both the pegasus and the dashboard I18n backends (which has some interesting implications for our fallback logic)

**Comment by Hamms** [comment]
closing in favor of https://github.com/code-dot-org/code-dot-org/pull/32607

**Pull request closed without merging**